Week 5 cluedo

monkeybusiness101 opened the pull request.

monkeybusiness101 committed:
added object for all suspects

monkeybusiness101 committed:
added suspects, weapons, rooms

monkeybusiness101 committed:
added arrays suspects, weapons, rooms

monkeybusiness101 committed:
completed iterations and blue level

monkeybusiness101 committed:
updated readme

monkeybusiness101 committed:
added styling to cards

monkeybusiness101 committed:
added favoriteWeapon to the different suspect arrays

monkeybusiness101 committed:
added shuffle of favouriteweapon when clicking suspect card

monkeybusiness101 committed:
comments and color button

jespa374 commented on code/clue.js:
```diff
@@ -3,33 +3,181 @@
-  mrsWhite
-  // ...  and the rest
+  mrsWhite,
+  mrPlum,
+  mrsPeacock,
+  mrMustard,
+  missScarlet
 ]
 
-const weapons = []
+const weapons = [
+  rope,
+  knife,
+  candlestick,
+  dumbbell,
+  poison,
+  axe,
+  bat,
+  trophy,
+  pistol
+]
 
```
Not sure if it's preferred to write array items vertically or horizontally but I think it saves a lot of space if you write as a row if there are not so many items like in the rooms array.

EmOlsz replied:
I would agree with Max over here - it's far more clean to format array in a away that single element has separate line

jespa374 commented on code/clue.js:
```diff
@@ -41,19 +189,73 @@ const randomSelector = array => {
 // With a killer, a weapon and a room.
 // The values will be set later.
 
+let mystery = {
+  killer:'',
+  weapon:'',
+  room:''
+}
```
Any reason for using let instead of const? It's recommended to use const as much as possible if there is not a specific reason for using let.

EmOlsz replied:
Good point Jessica!
Moreover, super small thing, but I would suggest you to initialise your mystery object with null values rather than empty strings. It's about sending proper information for other developers : null value indicates that probably we will replace these values with something else, but without saying what type will that new value have for now. Initialising your object with empty string values ' ' could be misunderstood by some one in a way that your properties will later on by replaced by new strings only. As a sum up, this would be better :
```
const mystery = {
  killer: null,
  weapon: null,
  room: null,
}
```

monkeybusiness101 replied:
fixed

jespa374 commented on code/clue.js:
```diff
@@ -41,19 +189,73 @@ const randomSelector = array => {
   mystery.killer = randomSelector(suspects)
+  
+  // Shuffle weapon function
+  shuffleFavouriteWeapon = () => {
+    mystery.killer.favouriteWeapon = randomSelector(weapons); 
+  };
```
You are using semicolons here but everywhere else you are not using semicolon. Be consistent.

EmOlsz replied:
Once again, good job Jessica!
One more thing to point out - in line 208 there is a declaration of variable without any keyword `let` or `const`. Because of that you've created global variable `shuffleFavouriteWeapon`. Briefly speaking, of you would create `const shuffleFavouriteWeapon = ...`, it would be scoped to in `pickKiller` function only and unavailable from outer scope. Because you omitted `const`, your variable somehow "slipped out" from `pickKiller` function.
This is huge antipattern, we should never (well, in 99.99% cases being honest) create global variables. Please, fix that asap

monkeybusiness101 replied:
Thank you for that Maks! I added const to the function. I also moved the function out of pickKiller. I don't know if that was what you wanted but perhaps makes the function more useful?

EmOlsz replied:
Probably it is, good idea :)

jespa374 commented on code/clue.js:
```diff
@@ -41,19 +189,73 @@ const randomSelector = array => {
+    mystery.killer.favouriteWeapon = randomSelector(weapons); 
+  };
+    shuffleFavouriteWeapon();
+    console.log(mystery.killer.favouriteWeapon.name)
```
No need of this console.log right? Maybe you can remove it.

EmOlsz replied:
👍

jespa374 commented on code/clue.js:
```diff
@@ -41,19 +189,73 @@ const randomSelector = array => {
   ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
+  document.getElementById('killerFavoriteWeapon').innerHTML = `${mystery.killer.favouriteWeapon.name}`
 }
+document.getElementById('killerCard').addEventListener('click', pickKiller)
```
Another way to write this would be `document.getElementById('killerCard').onclick = pickKiller`

EmOlsz replied:
Yes, but it's alternative, you can choose whatever way you want

jespa374 commented on code/index.html:
```diff
@@ -7,13 +7,20 @@
         <p id="killerAge"></p>
         <p id="killerOccupation"></p>
+        <p id="killerFavoriteWeapon"></p>
         <img id="killerImage" src="" height="100" />
```
Add an alt attribute to the img in case the picture isn't visible and for accessibility

EmOlsz replied:
Very nice Jessica

monkeybusiness101 replied:
fixed

jespa374 reviewed (approved):
Great job Max! :) You have done the blue level steps and also achieved red level with the favourite shuffle weapon. Congrats! I think you write clean code and the comments really help the viewer. Apart from the points mentioned in the code you can probably style the page more and also try to make the page more responsive in case you want to improve it but overall it's well done  👍

EmOlsz commented on code/clue.js:
```diff
@@ -3,33 +3,181 @@
   firstName: 'Jacob',
   lastName: 'Green',
-  color: 'green',
+  color: 'Green',
```
`color` property of suspects objects should be written lowercase - CSS expect from us lowercase values of colours

EmOlsz commented on code/clue.js:
```diff
@@ -41,19 +189,73 @@ const randomSelector = array => {
+
+const revealMystery = () => {
+  document.getElementById('mystery').innerHTML = `The murder was committed by ${mystery.killer.firstName} ${mystery.killer.lastName}, in the ${mystery.room} with a ${mystery.weapon.name}`
+}
```
You forgot to block suspect, weapon and room cards functionality over here - even after revealing crime I was able to change suspect card

EmOlsz commented on code/style.css:
```diff
@@ -1,6 +1,15 @@
   margin: 0;
   font-family: "Courier New", Courier, monospace;
+  
+
```
Unnecessary extra blank line

EmOlsz commented on code/style.css:
```diff
@@ -9,20 +18,26 @@ p {
   height: 600px;
-  background: #ccc;
+  background: #093028;  /* fallback for old browsers https://cssgradient.io/gradient-backgrounds/ */
+  background: -webkit-linear-gradient(to right, #237A57, #093028);  /* Chrome 10-25, Safari 5.1-6 https://cssgradient.io/gradient-backgrounds/*/
```
Very good Max :)

EmOlsz reviewed (changes requested):
Congrats Max, good job this week :) Take a look on both Jessica's and my comments and fix stuff we pointed out. I would especially like you to change unwanted global variable (there is a comment in code about that).
After changes are applied, I will approve your project!

monkeybusiness101 commented:
@EmOlsz Hi Maks! I have fixed the global variable and moved it out of pickKiller. I have also fixed, indentation, img alt, changed mystery object to null

monkeybusiness101 committed:
fixed indentation, alt image, mystery object to null, added const to shufflefavouriteweapon and moved it out of pickKiller

monkeybusiness101 committed:
removed weird files from lab
